api/src/main/java/org/openmrs/layout/name/NameSupport.java

```diff
@@ -30,13 +43,98 @@ public static NameSupport getInstance() {
 		if (singleton == null) {
 			throw new APIException("Not Yet Instantiated");
 		} else {
+			singleton.init();
 			return singleton;
 		}
 	}
+
+	/**
+	 * Initializes layout templates with a custom template configured
+	 * via the "layout.name.template" GP.
+	 */
+	private void init() {
+		if (initialized) {
+			return;
+		}
+		Context.getAdministrationService().addGlobalPropertyListener(singleton);
+		// Get configured name template to override the existing one if any
+		String layoutTemplateXml = Context.getAdministrationService().getGlobalProperty(
+			OpenmrsConstants.GLOBAL_PROPERTY_LAYOUT_NAME_TEMPLATE);
+		NameTemplate nameTemplate = deserializeXmlTemplate(layoutTemplateXml);
+		
+		if (nameTemplate != null) {
```

Comment:
If nameTemplate turns out to be null, would this mean that the init() method has not yet been called and hence initialized = false?

Reply:
@Ruhanga did you see this? https://openmrs.atlassian.net/browse/TRUNK-6225